### DISCUSSION TITLE
Chronological postfix.

### DISCUSSION THREAD

**Discussion by maxime-rey** [retorquere/zotero-better-bibtex#2891] [Q&A] [answered]
Hi all, I have a question regarding postfix in citation keys, as I just realised that they are not in chronological order.
I use the citation key formula "[auth][year][postfix+1=%(a)s]", and I just realised that a paper by xxx on July 2018 was xxx2018b, while the one from October was xxx2018a. 
Is there a way to change this behaviour? I assumed that if we used [year], it should obviously postfix in chronological order.

Thanks in advance for the help!

**Comment by retorquere** [comment]
Not really, no, other than manually editing them. The postfix just makes them unique, and if I'd change them based on date and you had already cited the paper, your article would suddenly be citing the wrong paper.

**Reply by maxime-rey** [reply]
Thank you for the fast answer!  
Looking at this the other way around, if there are doubles when publishing a paper, the order should be chronological. Then, if one uses better BibTeX writing it, the naming scheme from the paper will not correspond to the citation keys made by better BibTeX, which is very error-prone.  

To avoid this, I was using chronological citation keys ordered by hand. I'm currently trying to switch to better BibTeX, but if I do so, all the references will be wrong since it's not chronological. How is the current order decided ?

**Reply by retorquere** [reply, marked as answer]
Ah, bulk assignment is different. Yes, that is possible, even if it is a bit roundabout.

1. Change the citekey formula into something you *don't* want (simply a string like `'dummy'` will do)
2. select all items you want changed, right-click -> better bibtex -> refresh
3. change the citekey formula you do want (maybe the default `auth.lower + shorttitle(3,3) + year`, maybe something else)
4. order the items chronologically by adding the `Date` column to the middle pane and sorting on it
5. select the top item you want changed and shift-click the last item you want changed
6. right-click -> better bibtex -> refresh

**Reply by maxime-rey** [reply]
That's exactly what I needed, thanks!
